handlers/config.go

```diff
@@ -49,25 +47,25 @@ type DeploymentConfig struct {
 // PublicConfig is a subset of Kiali configuration that can be exposed to clients to
 // help them interact with the system.
 type PublicConfig struct {
-	AccessibleNamespaces []string                    `json:"accessibleNamespaces,omitempty"`
-	AuthStrategy         string                      `json:"authStrategy,omitempty"`
-	AmbientEnabled       bool                        `json:"ambientEnabled,omitempty"`
-	ClusterInfo          ClusterInfo                 `json:"clusterInfo,omitempty"`
-	Clusters             map[string]business.Cluster `json:"clusters,omitempty"`
-	Deployment           DeploymentConfig            `json:"deployment,omitempty"`
-	GatewayAPIEnabled    bool                        `json:"gatewayAPIEnabled,omitempty"`
-	HealthConfig         config.HealthConfig         `json:"healthConfig,omitempty"`
-	InstallationTag      string                      `json:"installationTag,omitempty"`
-	IstioAnnotations     IstioAnnotations            `json:"istioAnnotations,omitempty"`
-	IstioCanaryRevision  IstioCanaryRevision         `json:"istioCanaryRevision,omitempty"`
-	IstioStatusEnabled   bool                        `json:"istioStatusEnabled,omitempty"`
-	IstioIdentityDomain  string                      `json:"istioIdentityDomain,omitempty"`
-	IstioNamespace       string                      `json:"istioNamespace,omitempty"`
-	IstioLabels          config.IstioLabels          `json:"istioLabels,omitempty"`
-	IstioConfigMap       string                      `json:"istioConfigMap"`
-	KialiFeatureFlags    config.KialiFeatureFlags    `json:"kialiFeatureFlags,omitempty"`
-	LogLevel             string                      `json:"logLevel,omitempty"`
-	Prometheus           PrometheusConfig            `json:"prometheus,omitempty"`
+	AccessibleNamespaces []string                      `json:"accessibleNamespaces,omitempty"`
+	AuthStrategy         string                        `json:"authStrategy,omitempty"`
+	AmbientEnabled       bool                          `json:"ambientEnabled,omitempty"`
+	ClusterInfo          ClusterInfo                   `json:"clusterInfo,omitempty"`
+	Clusters             map[string]kubernetes.Cluster `json:"clusters,omitempty"`
```

Comment:
This is sort of confusing.  Not sure why we need `ClusterInfo` just to replicate the home cluster info when there are multiple clusters.  Instead of being backward compatible maybe we should clean this up.   We could maybe defer until we do more work on the `Mesh` page, because I think in the future we want to show that menu option at all times, even for single-cluster, and migrate the detailed control plane info to there from `Overview`.

Reply:
That's a good point. I just kept the `ClusterInfo` field around but that same info is exposed by the `clusters` field. I'll look at consolidating them as part of this change.

Reply:
The rebase on this was kind of hairy and this PR is already large. If it's alright I'll create a separate issue for removing the `ClusterInfo`.